public/scripts/infrastructure/views/ingredient-list-view.js

```diff
@@ -12,27 +13,39 @@ function saveAndFireListCleared(aSet, olList) {
     triggerListCleared(olList, currentElements);
 }
 
+/**
+ * 
+ * @param {string} ingredientName 
+ * @returns {string}
+ */
+function valueToId(ingredientName) {
+    return ingredientName.split(' ').join('-').toLowerCase();
+}
+
 /**
  * 
  * @param {string} ingredientName 
  * @returns {HTMLLIElement}
  */
 function createSelectableListItem(ingredientName) {
     const listEl = document.createElement('li');
+    listEl.ariaSelected = false;
+    listEl.id = `${valueToId(ingredientName)}-item`;
+    listEl.tabIndex = -1;
+    listEl.setAttribute('role', 'option');
 
     const checkBoxInput = document.createElement('input');
     checkBoxInput.type = 'checkbox';
     checkBoxInput.name = 'selected-ingredients';
-    checkBoxInput.ariaChecked = false;
-    checkBoxInput.id = ingredientName.replace(' ', '-').toLowerCase();
     checkBoxInput.value = ingredientName;
+    checkBoxInput.id = valueToId(ingredientName);
+    checkBoxInput.tabIndex = -1;
 
     const label = document.createElement('label');
     label.htmlFor = checkBoxInput.id;
-    label.textContent = ingredientName;
```

Comment:
This gives focus to the input associated with that label instead of the list element containing it, trapping the user from moving through the options.